### PR TITLE
SIMD-0084: Disable rent fees collection

### PR DESCRIPTION
**Pull request by HaoranYi**
Add simd for disable rent fees collection.

### CODE REVIEW

**Comment by apfitzge** [comment]
Without rent payments, can we eliminate the concept & term of "rent" entirely? 

- How confident are we with the current minimum balance calculation(s)? If we remove rent, it seems this would become more difficult to change in the future.
- If the concept of rent goes away, should the `Rent` sysvar continue to be available?

**Comment by HaoranYi** [comment]
Yeah. We still want to enforce that all new accounts are rent exempt.

**Comment by apfitzge** [comment]
> This only eliminates rent collection (and distribution), so the concept of rent still remains. We could rename it to "Minimum Balance", since that's what it effectively will be now.

Right I was essentially asking this. If no one pays "rent", it's not really "rent" but a minimum required balance. I was definitely not suggesting we get rid of the minimum balance.


That still leaves my other question about if we're comfortable with the current minimum balances we have now.
Right now, if we decided the minimum balances were not sufficient to cover storage costs, we could increase these. Some accounts would become rent-paying again. They'd eventually get drained and we'd go back to not having rent-paying accounts.
But if we get rid of rent-paying, then it's more difficult for us to make this increase.
So we should be confident that the current values are correct.

Does that make sense?

**Comment by HaoranYi** [comment]
With disks going cheaper and cheaper in future, I think, most likely, we will only lower the minimum required balance in future.

File: proposals/0084-disable-rent-fees-collection.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+## Summary
+
+Add a new feature to disable rent fees collections.
+
+## Motivation
+
+The total number of rent paying accounts on solana network is phasing out. And
+it is no longer possible to create new rent paying account on the network.
+Therefore, when all the rent paying accounts are gone in the network, we would
+like to disable rent fee collection on the network through a feature.
```
Comment (lheeger-jump) [inline proposals/0084-disable-rent-fees-collection.md:22]: The number of rent paying accounts is about 220K as of a few months ago and the last one will be drained of lamports in about one year.

**Review by lheeger-jump — CHANGES_REQUESTED** [review]

File: proposals/0084-disable-rent-fees-collection.md (hunk 1 above)
Comment (HaoranYi) [reply proposals/0084-disable-rent-fees-collection.md:22]: As of today, there are no more rent-paying accounts on MNB. All these rent paying accounts have already been topped off to be rent-exempt.

File: proposals/0084-disable-rent-fees-collection.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+None
+
+## New Terminology
+
+None
+
+## Detailed Design
+
+When the feature - "disable rent fees collections" is activated, rent will no
+longer be collected from accounts and distributed to validators.
```
[the remark this answers is not in the document]
Comment (HaoranYi) [reply proposals/0084-disable-rent-fees-collection.md]: yeah. added.

File: proposals/0084-disable-rent-fees-collection.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+1. Accounts, which are not "rent exempt", will no longer pay rents. And
+   validators will not be paid any rent fees. (Both of these should already not
+   happen when all rent paying accounts are gone.)
+
+2. Other implementations of validator client will not need to implement rent
+   processing.
+
+3. The performance of validators will be better since there is no more rent
+   processing.
```
[the remark this answers is not in the document]
Comment (HaoranYi) [reply proposals/0084-disable-rent-fees-collection.md]: yeah. removed.

File: proposals/0084-disable-rent-fees-collection.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+## Detailed Design
+
+When the feature - "disable rent fees collections" is activated, rent will no
+longer be collected from accounts nor will it be distributed to validators.
+
+Note that this does **not** change the requirement that existing rent-paying
+accounts, which need to be made rent-exempt first before any withdrawals can be
+make from them.
+
+The PR for this work is at https://github.com/solana-labs/solana/pull/33945
```
[the remark this answers is not in the document]
Comment (HaoranYi) [reply proposals/0084-disable-rent-fees-collection.md]: Removed.

File: proposals/0084-disable-rent-fees-collection.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+   processing.
+
+1. The performance of validators will be better since there is no more rent
+   processing.
+
+## Security Considerations
+
+There will be no more rent paying accounts once the feature is activated. It is
+already impossible to create any new rent paying accounts on the network, there
+should be no security issue. However, if for some reason rent paying accounts
```
[the remark this answers is not in the document]
Comment (HaoranYi) [reply proposals/0084-disable-rent-fees-collection.md:54]: This feature didn't stop *rewriting* account. The bank hash will still capture the accounts' hashes for the rewritten accounts.

File: proposals/0084-disable-rent-fees-collection.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+None
+
+## New Terminology
+
+None
+
+## Detailed Design
+
+When the feature - "disable rent fees collections" is activated, all accounts
+will be treated as "Rent Exempt".
```
[the remark this answers is not in the document]
Comment (HaoranYi) [reply proposals/0084-disable-rent-fees-collection.md]: Updated.

File: proposals/0084-disable-rent-fees-collection.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+None
+
+## Security Considerations
+
+None
+
+## Backwards Compatibility
+
+Not applicable.
```
[the remark this answers is not in the document]
Comment (HaoranYi) [reply proposals/0084-disable-rent-fees-collection.md]: Yeah. Updated.

File: proposals/0084-disable-rent-fees-collection.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+## Detailed Design
+
+When the feature - "disable rent fees collections" is activated, all accounts
+will be treated as "Rent Exempt".
+
+The PR is at https://github.com/solana-labs/solana/pull/33945
+
+## Impact
+
+None
```
[the remark this answers is not in the document]
Comment (HaoranYi) [reply proposals/0084-disable-rent-fees-collection.md]: Updated.

**Review by lheeger-jump — DISMISSED** [review]
I think this change looks great. If we can get an approval from Anza, we can get it merged ASAP, as it does not have any new comments.

**Comment by HaoranYi** [comment]
@brooksprumo @jeffwashington What do you think about this SIMD? Can we merge it?

File: proposals/0084-disable-rent-fees-collection.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+## Detailed Design
+
+When the feature - "disable rent fees collections" is activated, rent will no
+longer be collected from accounts nor will it be distributed to validators.
+
+Note that this does **not** change the requirement that existing rent-paying
+accounts, which need to be made rent-exempt first before any withdrawals can be
+make from them.
```
[the remark this answers is not in the document]
Comment (HaoranYi) [reply proposals/0084-disable-rent-fees-collection.md]: These are good observations and notes. But I don't think these points are relevant to the simd design. I prefer not to include them in the content of the SIMD.

**Comment by HaoranYi** [comment]
@jacobcreech We have received approval for this SIMD from FD and Anza. Do you mind merging it?

**Comment by 0xSol** [comment]
Seem this SIMD reaches consensus. All please take a final check, and if no objections commented by Friday May 10, will merge.  Thanks for great collaboration!

**Review by lheeger-jump — APPROVED** [review]

**Pull request merged**